pkg/controllers/capr/bootstrap/controller.go

```diff
@@ -507,12 +509,20 @@ func (h *handler) reconcileMachinePreTerminateAnnotation(bootstrap *rkev1.RKEBoo
 				}
 			}
 		}
+
+		// Although we want to make sure no nodes are using this machine's join-url, if plans are currently being reconciled
+		// they may not have successfully applied their new plans yet.
+		if !capr.Reconciled.IsTrue(cp) {
```

Comment:
Is this the condition we want to use...? 

This condition indicates that the RKEcontrolplane spec vs. applied spec is not equal, but in the case of scaling down, the spec won't change.

Reply:
If you are trying to make sure that the cluster's system-agents have fully reconciled I think this needs to be changed.

Even then, I think there are edge cases when we don't actually care whether a system-agent has reconciled or not, for example, think about a worker node that may be shut down.

Maybe we remove this for now and file an issue to revisit it more holistically after we give it some thought.

Reply:
Let me test without this to make sure, if the issue doesn't manifest I'll just remove it.